src/search_algo.hpp

```diff
@@ -1133,7 +1134,48 @@ inline void _performAlignment(TDepSetH &      depSetH,
 }
 
 template <typename TLocalHolder>
-inline void iterateMatchesFullSimd(TLocalHolder & lH, bsDirection const dir = bsDirection::fwd)
+inline void _widenAndPreprocessMatches(std::span<Match> & matches, TLocalHolder & lH)
+{
+    auto before = matches.size();
+
+    for (Match & m : matches)
+        _widenMatch<TLocalHolder>(m, lH);
+
+    std::ranges::sort(matches);
+
+    if (matches.size() > 1)
+    {
+        // pairwise merge from left to right
+        for (auto it = matches.begin(); it < matches.end() - 1; ++it)
+        {
+            Match & l = *it;
+            Match & r = *(it + 1);
+            if ((std::tie(l.qryId, l.subjId) == std::tie(r.qryId, r.subjId)) && (l.subjEnd >= r.subjStart))
+            {
+                l.subjEnd   = r.subjEnd;
+                r.subjStart = l.subjStart;
+            }
+        }
+
+        // pairwise "swallow" from right to left
+        for (auto it = matches.rbegin(); it < matches.rend() - 1; ++it)
+        {
+            Match & l = *it;
```

Comment:
Given that the comment says from right to left, I understood it in a way that we are still "looking" at the vector the same way as before. However, now that we are backwards iterating technically ```l``` is to the right of ```r``` in the sense of that comment and it confused me naming-wise when I tried to understand what is happening (because it does not match the ```l``` and ```r``` in the loop before order-wise). Can we find a different way of naming this that is more intuitive?

Reply:
That's a good point. I have switched `r` and `l`. Is that OK?

Reply:
yes, thanks!